### ISSUE TITLE
Test Payara on the IBM JDK 7 and 8

### ISSUE THREAD

**Issue by smillidge** [payara/Payara#248]
Payara needs to be tested on IBM JDK 7 and 8. There were know issues in the past with GlassFish on IBM JDK.

**Comment by smillidge** [comment]
Proprietary APIs need to be removed. GlassFish top level JIRA references proprietary api problems.
https://java.net/jira/browse/GLASSFISH-19812

**Comment by smillidge** [comment]
It could be best to use bouncy castle as the security provider and replace the internal apis with their bouncy castle equivalents.

**Comment by mikecroft** [comment]
There is a related question on StackOverflow about problems running GlassFish 4.1 on AIX. This may not necessarily be a problem with GlassFish, though.
http://stackoverflow.com/questions/29623539/is-glassfish-4-1-supported-in-aix-6-1-0-0/29637183

**Comment by smillidge** [comment]
EJB container fails to initialise on IBM JDK when deploying CargoTracker. Looks like there are dependencies on internal Oracle JDK security apis. These could likely be replaced with bouncy castle apis. 

```java
Caused by: java.lang.NoClassDefFoundError: sun.security.x509.X509CertImpl

	at com.sun.enterprise.iiop.security.SecurityIIOPInterceptorFactory.getServerInterceptorInstance(SecurityIIOPInterceptorFactory.java:155)

	at com.sun.enterprise.iiop.security.SecurityIIOPInterceptorFactory.createServerRequestInterceptor(SecurityIIOPInterceptorFactory.java:108)

	at org.glassfish.enterprise.iiop.impl.GlassFishORBInitializer.post_init(GlassFishORBInitializer.java:134)

	at com.sun.corba.ee.impl.interceptors.PIHandlerImpl.postInitORBInitializers(PIHandlerImpl.java:917)

	at com.sun.corba.ee.impl.interceptors.PIHandlerImpl.initialize(PIHandlerImpl.java:267)

	at com.sun.corba.ee.impl.orb.ORBImpl.postInit(ORBImpl.java:636)

	at com.sun.corba.ee.impl.orb.ORBImpl.set_parameters(ORBImpl.java:697)

	at com.sun.corba.ee.impl.orb.ORBImpl.setParameters(ORBImpl.java:684)

	at com.sun.corba.ee.spi.osgi.ORBFactory.initialize(ORBFactory.java:107)

	at org.glassfish.enterprise.iiop.impl.GlassFishORBManager.initORB(GlassFishORBManager.java:585)

	at org.glassfish.enterprise.iiop.impl.GlassFishORBManager.getORB(GlassFishORBManager.java:264)

	at org.glassfish.enterprise.iiop.impl.GlassFishORBFactoryImpl.createORB(GlassFishORBFactoryImpl.java:93)

	at org.glassfish.enterprise.iiop.api.GlassFishORBHelper.getORB(GlassFishORBHelper.java:163)

	at org.glassfish.enterprise.iiop.api.GlassFishORBHelper.getProtocolManager(GlassFishORBHelper.java:231)

	at com.sun.ejb.containers.BaseContainer.initializeProtocolManager(BaseContainer.java:898)

	... 71 more
```

**Comment by smillidge** [comment]
3.1.2.2 uses explicitly patched jars that replace sun.security.util.* apis with the equivalent com.ibm.security apis

**Comment by smillidge** [comment]
Fixed by #358 in payara-blue branch. Download will be released with 4.1.153
